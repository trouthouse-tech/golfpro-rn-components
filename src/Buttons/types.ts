export type ButtonProps = {
  onPress(): void;
  text: string;
  buttonColor?: string;
  textColor?: string;
  borderColor?: string;
  additionalStyling?: {};
};
