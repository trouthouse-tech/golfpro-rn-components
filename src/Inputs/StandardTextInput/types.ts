import {KeyboardType} from 'react-native';

export type InputProps = {
  onChangeText(text: string): void;
  value: string;
  autoFocus?: boolean;
  secureTextEntry?: boolean;
  placeholder?: string;
  onBlur?(): void;
  keyboardType?: KeyboardType;
  editable?: boolean;
  blurOnSubmit?: boolean;
};
