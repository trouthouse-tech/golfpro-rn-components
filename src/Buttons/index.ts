import LargeHallowSquareOnPress from './LargeHallowSquareOnPress';
import LargeButton from './LargeButton';
import LargeSquareOnPress from './LargeSquareOnPress';
import TextButton from './TextButton';
import SmallHallowButton from './SmallHallow';
import SmallHallowRoundedButton from './SmallHallowRoundedButton';
import MediumButton from './MediumButton';
import SmallButton from './SmallButton';

export {
  LargeHallowSquareOnPress,
  LargeButton,
  LargeSquareOnPress,
  TextButton,
  SmallButton,
  SmallHallowButton,
  SmallHallowRoundedButton,
  MediumButton,
};
