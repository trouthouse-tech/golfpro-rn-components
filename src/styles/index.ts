import * as Fonts from './fonts';
import * as Utils from './util';
import * as Colors from './colors';

export {Fonts, Utils, Colors};
