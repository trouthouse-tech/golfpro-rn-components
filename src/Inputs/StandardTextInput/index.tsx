import React from 'react';

import { StyleSheet, TextInput } from 'react-native';
import { Fonts, Utils } from '../../styles';
import { InputProps } from './types';

export function StandardTextInput(props: InputProps) {
    const {
        placeholder,
        onChangeText,
        value,
        autoFocus,
        secureTextEntry,
        onBlur,
        keyboardType,
        editable,
        blurOnSubmit = false,
    } = props;

    return (
        <TextInput
            placeholder={placeholder}
            placeholderTextColor="rgba(0,0,0,.5)"
            autoCapitalize="none"
            autoFocus={autoFocus || false}
            secureTextEntry={secureTextEntry || false}
            style={styles.textInput}
            onChangeText={(username) => onChangeText(username)}
            value={value}
            onBlur={() => !onBlur || onBlur()}
            keyboardType={keyboardType || 'default'}
            editable={editable || true}
            blurOnSubmit={blurOnSubmit}
        />
    );
}

const styles = StyleSheet.create({
    textInput: {
        fontSize: Fonts.normal,
        borderWidth: 1,
        borderColor: 'gray',
        marginVertical: 10,
        borderRadius: 10,
        width: Utils.DEVICE_WIDTH / 1.3,
        height: 40,
        padding: 10,
        color: 'black',
    },
});
