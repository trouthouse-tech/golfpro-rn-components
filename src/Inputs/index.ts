import {StandardTextInput} from './StandardTextInput/';
import {InputProps} from './StandardTextInput/types';

export {StandardTextInput};

export type {InputProps};
