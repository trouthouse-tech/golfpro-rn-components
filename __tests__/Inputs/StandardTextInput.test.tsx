import React from 'react';
import { StandardTextInput } from "../../src/Inputs";
import { InputProps } from "../../src/Inputs/StandardTextInput/types";

import { render, cleanup, fireEvent } from "@testing-library/react-native";
import '@testing-library/jest-native/extend-expect';
import renderer from 'react-test-renderer';

describe('StandardTextInput', () => {

    const mockFunction = jest.fn();

    const defaultInputProps: InputProps = {
        value: "Name",
        autoFocus: true,
        editable: true,
        placeholder: "Your Name?",
        secureTextEntry: true,
        onBlur: mockFunction,
        onChangeText: mockFunction,
    }

    afterEach(cleanup);

    test('renders and matches snapshot', () => {
        const tree = renderer.create(<StandardTextInput {...defaultInputProps} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

});