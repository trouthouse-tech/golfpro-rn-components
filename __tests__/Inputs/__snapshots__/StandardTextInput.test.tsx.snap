// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StandardTextInput renders and matches snapshot 1`] = `
<TextInput
  allowFontScaling={true}
  autoCapitalize="none"
  autoFocus={true}
  editable={true}
  keyboardType="default"
  onBlur={[Function]}
  onChangeText={[Function]}
  placeholder="Your Name?"
  placeholderTextColor="rgba(0,0,0,.5)"
  rejectResponderTermination={true}
  secureTextEntry={true}
  style={
    Object {
      "borderColor": "gray",
      "borderRadius": 10,
      "borderWidth": 1,
      "color": "black",
      "fontSize": 18,
      "height": 40,
      "marginVertical": 10,
      "padding": 10,
      "width": 576.9230769230769,
    }
  }
  underlineColorAndroid="transparent"
  value="Name"
/>
`;
